core/cat/looking_glass/recall_settings.py

```diff
@@ -0,0 +1,31 @@
+"""Module for retrieving default configurations for episodic, declarative and procedural memories"""
+
+
+class RecallSettings:
```

Comment:
Hi @scicco having a class is a good idea, but this one is a little over engineered imho.
Try to have a simple class extending `cat.utils.BaseModelDict` so it can be used both as an object (leveraging a pydantic models with types and defaults) and as a dict and will be retrocompatible.
Examples [here](https://github.com/cheshire-cat-ai/core/blob/f14f2c0acebddd3f1f1efb6ba3b161bea45e9e3a/core/cat/convo/messages.py#L39)

```python
class RecallSettings(BaseModelDict):
    embedding
    k
    threshold
    # skipping types and defaults because I'm cooked
    ...
```

Then when it is time to use it:

```python
default_episodic_recall_config = RecallSettings(
    embedding=recall_query_embedding,
    user_id=self.user_id
)
```

Reply:
done